### PR TITLE
Security overview documentation

### PR DESCRIPTION
**Pull request by sthetland**
A minor-to-moderate revamp of the Security Docs. This change:

1. Pulls general security concepts from the Basic Auth extensions page and puts them into an overview security page. 
2. Adds best practices and sample setup steps for TLS, basic auth, and LDAP authorizer to the overview page. 
3. Gathers a few other existing security-related pages into a new Security page tree category under Operations.

### CODE REVIEW

File: docs/operations/security-overview.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
+---
+id: security-overview
+title: "Security overview"
+---
+
+
+## Overview
+
+By default, security features in Druid are disabled, that is, TLS is disabled and user authentication does not occur. To use these features, you need to configure security in Druid.
```
Comment (2bethere) [inline docs/operations/security-overview.md:9]: "By default, security features in Druid are disabled to simplify initial development experience. However, security features must be configured in a production deployment. Those features including TLS, authentication, and authorization".

File: docs/operations/security-overview.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+## Overview
+
+By default, security features in Druid are disabled, that is, TLS is disabled and user authentication does not occur. To use these features, you need to configure security in Druid. 
+
+This document gives you an overview of security features in Druid and how to configure them, and some best practices for securing Druid.
+
+
+## Best practices
+
+* Do not expose the Druid Console on an untrusted users or networks. Access to the console effectively confers access the file system on the installation machine, via file browsers in the UI. You should use an API gateway that restricts who can connect from untrusted networks, whitelists the specific APIs that your users need to access, and implements account lockout and throttling features.
```
Comment (2bethere) [inline docs/operations/security-overview.md:16]: "Do not expose the Druid Console without authentication on untrusted networks."

File: docs/operations/security-overview.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+* Grant users the minimum permissions necessary to perform their functions. For instance, do not allow user who only need to query data to write to data sources or view state.  
+* Disable JavaScript, as noted in the [Security section](https://druid.apache.org/docs/latest/development/javascript.html#security) of the JavaScript guide.
+* Run Druid as an unprivileged Unix user on the installation machine (not root).
```
Comment (2bethere) [inline docs/operations/security-overview.md:19]: Administrator users on Druid have the same permission of the unix user account it is running under. If the Druid process is running under root user account in the OS, then Administrator users on Druid can read/write all files that the root account has access to, including

Comment (2bethere) [reply docs/operations/security-overview.md:19]: There needs to be a warning box around this one. It is very very important not to do this.

File: docs/operations/security-overview.md [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+   druid.escalator.authorizerName=MyBasicMetadataAuthorizer
+
+   druid.auth.authorizers=["MyBasicMetadataAuthorizer"]
+
+   druid.auth.authorizer.MyBasicMetadataAuthorizer.type=basic
+   ```
+
+3. Restart the cluster. 
+
+See [Basic Security](../development/extensions-core/druid-basic-security) for more information. For more on authentication extensions, see [Kerberos](../development/extensions-core/druid-kerberos), [Authentication and Authorization](../design/auth), and [Authentication and Authorization](../design/auth)
```
Comment (jon-wei) [inline docs/operations/security-overview.md:102]:
```suggestion
See [Authentication and Authorization](../design/auth) for more information about the Authenticator, Escalator, and Authorizer concepts. See [Basic Security](../development/extensions-core/druid-basic-security) for more information about the extension used in the examples above, and [Kerberos](../development/extensions-core/druid-kerberos) for Kerberos authentication.
```

File: docs/operations/security-overview.md [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+After enabling the basic auth extension, you can add users, roles, and permissions via the Druid Coordinator `user` endpoint. Note that you cannot assign permissions directly to individual users. They must be assigned through roles. 
+
+The following diagram depicts the authorization model, and the relationship between users, roles, permissions, and resources.
+ 
+![Druid Security model](../assets/security-model-2.png "Druid security model") 
+
+
+The following steps walk through a sample setup procedure:  
+
+> The Coordinator API port is 8081 for non-TLS connections and 8281 for secured connections.
```
Comment (jon-wei) [inline docs/operations/security-overview.md:116]:
```suggestion
> The default Coordinator API port is 8081 for non-TLS connections and 8281 for secured connections.
```

Comment (jon-wei) [reply docs/operations/security-overview.md:116]: Would also suggest mentioning that if TLS is enabled, the user may have to adjust the example curl commands to account for that (e.g., they used a self-signed certificate for the Druid servers that's not trusted by curl by default)

File: docs/operations/security-overview.md [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+      "name": "STATE",
+      "type": "STATE"
+    },
+    "action": "READ"
+    }
+    ]
+    ```
+
+
+## Configuring an LDAP authorizor
```
Comment (jon-wei) [inline docs/operations/security-overview.md:170]:
```suggestion
## Configuring an LDAP authenticator
```

File: docs/operations/security-overview.md [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+
+As an alternative to using the basic metadata authenticator, as shown in the previous section, you can use LDAP to authorize users. The following steps provide an overview of the setup steps. For more information on these settings, see [Properties for LDAP user authentication](https://druid.apache.org/docs/latest/development/extensions-core/druid-basic-security.html#properties-for-ldap-user-authentication).
```
Comment (jon-wei) [inline docs/operations/security-overview.md:172]:
```suggestion
As an alternative to using the basic metadata authenticator, as shown in the previous section, you can use LDAP to authenticate users. The following steps provide an overview of the setup steps. For more information on these settings, see [Properties for LDAP user authentication](https://druid.apache.org/docs/latest/development/extensions-core/druid-basic-security.html#properties-for-ldap-user-authentication).
```

File: docs/operations/security-user-auth.md [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+* STATE &ndash; Cluster-wide state resources.
+
+For specific resources associated with the types, see the endpoint list below and the corresponding descriptions of those endpoints on [API Reference](api-reference).
+
+There are two actions:
+
+* READ &ndash; Used for read-only operations.
+* WRITE &ndash; Used for operations that are not read-only.
+
+In practice, most deployments will only need to define to classes of users:
```
Comment (jon-wei) [inline docs/operations/security-user-auth.md:45]:
```suggestion
In practice, most deployments will only need to define two classes of users: 
```

File: docs/operations/security-overview.md [hunk 9]
```diff
@@ -0,0 +1,244 @@
+---
+id: security-overview
```
Comment (jon-wei) [inline docs/operations/security-overview.md:2]: This'll need the Apache license header

File: docs/operations/auth-ldap [hunk 10]
```diff
@@ -0,0 +1,165 @@
+---
+id: auth-ldap
+title: "Using LDAP auth"
```
Comment (jon-wei) [inline docs/operations/auth-ldap:3]: This'll need the Apache license header

File: docs/operations/security-overview.md [hunk 11]
```diff
@@ -0,0 +1,244 @@
...
+  druid.client.https.protocol=TLSv1.2
+  druid.client.https.trustStoreType=jks
+  druid.client.https.trustStorePath=imply-truststore.jks
+  druid.client.https.trustStorePassword=secret123  # replace with your own password
+
+  ``` 
+4. Add the `simple-client-sslcontext` extension to `druid.extensions.loadList` in common.runtime.properties. This enables TLS for Druid nodes acting as clients.
+5. Restart the cluster.
+
+For more information, see [TLS support](tls-support) and [Simple SSLContext Provider Module](../development/extensions-core/simple-client-sslcontext).
```
Comment (jon-wei) [inline docs/operations/security-overview.md:62]:
```suggestion
For more information, see [TLS support](tls-support.md) and [Simple SSLContext Provider Module](../development/extensions-core/simple-client-sslcontext). 
```

Comment (jon-wei) [reply docs/operations/security-overview.md:62]: This link and others will need `.md` at the end

**Review by jon-wei — APPROVED** [review]

**Pull request merged**